substrate/frame/staking/src/tests.rs

```diff
@@ -6849,6 +6849,102 @@ mod staking_interface {
 	}
 }
 
+mod staking_unsafe {
```

Comment:
We should have some tests to ensure `StakingInterface` impl consistency:
- existing getter-like functions work normally with virtual stakers (e.g. `stash_by_ctrl`, `total_stake`, etc.); `withdraw_unbonded` and `unbond` are already partially covered on the happy path, but it might be worth it to add a test case for invalid calls after a virtual bond as well, just to be safe
- functions which shouldn't work fail as expected (e.g. calling `StakingInterface::bond` after calling `<StakingInterface as StakingUnsafe>::virtual_bond`)

Reply:
I like the systematic thinking here, yes we should have these tests.